lib/eslint/eslint.js

```diff
@@ -765,6 +767,15 @@ class ESLint {
         return createRulesMeta(resultRules);
     }
 
+    /**
+     * Indicates if the given feature flag is enabled for this instance.
+     * @param {string} flag The feature flag to check.
+     * @returns {boolean} `true` if the feature flag is enabled, `false` if not.
+     */
+    hasFlag(flag) {
+        return privateMembers.get(this).options.flags.includes(flag);
+    }
```

Comment:
Question: The purpose of this is to provide mode detection? We'll always keep the list of all flags ever introduced and automatically enable them in future major versions?

Reply:
Should there also be a method for feature detection? This returns which flags are enabled, but not which flags are supported.

Reply:
Good questions:

1. My intent was that flags would be removed as features became stable or a new major release published. So if there's a `v10_config_lookup`, that would be available in v9 but would be removed in v10 because it would become the default.
2. What is the use case for a function that returns which flags are supported?

Reply:
What is the use case for the current implementation or `hasFlag`?

Reply:
There are two use cases:

1. To be able to inspect from outside of the `ESLint` class as to whether or not a flag is enabled. This is useful for integrations that may want to react differently depending on which flags are enabled.
2. To inspect from inside `ESLint`. Take the example of [the new config lookup RFC](https://github.com/eslint/rfcs/pull/120), which needs to change how configs are loaded based on the presence of a flag. `hasFlag()` means encapsulating the functionality in one place so it's easier to check (vs. pulling the list of flags or checking `linter`).